src/battle_main.c

```diff
@@ -3081,6 +3087,56 @@ void FaintClearSetData(void)
             gBattleMons[i].status2 &= ~(STATUS2_INFATUATED_WITH(gActiveBattler));
         if ((gBattleMons[i].status2 & STATUS2_WRAPPED) && *(gBattleStruct->wrappedBy + i) == gActiveBattler)
             gBattleMons[i].status2 &= ~(STATUS2_WRAPPED);
+
+        // If the fainted mon was holding another Pokemon in Sky Drop, release the mon and clear Sky Drop data
+        if (gActiveBattler == gBattleStruct->skyDropTargets[0] && i == gBattleStruct->skyDropTargets[1])
+        {
+            gBattleStruct->skyDropTargets[0] = 0xFF;
+            gBattleStruct->skyDropTargets[1] = 0xFF;
+            
+            gStatuses3[i] &= ~(STATUS3_SKY_DROPPED | STATUS3_ON_AIR);
```

Comment:
the status3 and status2 flags are cleared at the top of this function

Reply:
Status3 and status2 flags are only cleared for the fainted Pokémon (gActiveBattler), not for every Pokémon currently in battle. This means, if the attacker of a Sky Drop faints, it will be properly taken out of the semi-invulnerable state, but not the target. This section of code is to make sure the target is freed from Sky Drop and is removed from the in_air state.

Reply:
Good catch, thanks!